src/game/logic/object/locomotor.cpp

```diff
@@ -1174,13 +1231,184 @@ void Locomotor::Move_Towards_Position_Hover(
     }
 }
 
+bool Within_Epsilon(float f)
+{
+    return GameMath::Fabs(f) < 0.001f;
+}
+
+float Try_To_Rotate_Vector_3D(float max_angle, Vector3 &in_cur_dir, Vector3 &in_goal_dir, Vector3 &actual_dir)
+{
+    if (Within_Epsilon(max_angle)) {
+        actual_dir = in_cur_dir;
+        return 0.0f;
+    } else {
+        Vector3 v(in_cur_dir);
+        v.Normalize();
+        Vector3 v2(in_goal_dir);
+        v2.Normalize();
+        float f = GameMath::Acos(std::clamp(v * v2, -1.0f, 1.0f));
+
+        if (max_angle < 0.0f && (max_angle = -max_angle * f, Within_Epsilon(max_angle))) {
```

Comment:
move max angle out and put the epsilon check in its own case

Reply:
fixed.